### PR TITLE
Remove py36compat module

### PR DESCRIPTION
**Pull request by jhaigh0**
### Description of work

Removed the `py36compat` module since we're on python 3.10 now and cab use the built-in `dataclass` and `field` freely. 

*There is no associated issue.*

### To test:

- Code review and CI tests

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: scripts/SANS/sans/gui_logic/models/RowEntries.py [hunk 1]
```diff
@@ -68,7 +69,7 @@ def __eq__(self, other):
                 self.can_direct_period,
                 self.can_scatter_period,
                 self.can_transmission_period,
-                self.sample_direct,
+                self.sample_direct_period,
```
Comment (robertapplin) [inline scripts/SANS/sans/gui_logic/models/RowEntries.py:72]: Was this previously not working?

**Review by robertapplin — CHANGES_REQUESTED** [review]
This is a really nice tidy up of the code! The `py36compat` module is no longer needed because we use python 3.10 exclusively. I just had one quick question to help my understanding before I approve

File: scripts/SANS/sans/gui_logic/models/RowEntries.py (hunk 1 above)
Comment (jhaigh0) [reply scripts/SANS/sans/gui_logic/models/RowEntries.py:72]: well, `self.sample_direct` is in the comparison tuple twice, and `self.sample_direct_period` isn't in there so I think this is just a mistake. There is a test which tests a deep copy, it failed when I removed the `__eq__` override but passes with this correction.

Comment (robertapplin) [reply scripts/SANS/sans/gui_logic/models/RowEntries.py:72]: Ah I understand now, thanks for explaining!

**Review by robertapplin — APPROVED** [review]

**Review by peterfpeterson — APPROVED** [review]
The `py36compat` directory was deleted and imports were updated.

**Pull request merged**